etc/rust_analyzer_settings.json

```diff
@@ -0,0 +1,26 @@
+{
+    "rust-analyzer.rustc.source": "discover",
+    "rust-analyzer.linkedProjects": [
+        "Cargo.toml",
+        "cargo-miri/Cargo.toml",
+        "miri-script/Cargo.toml",
+    ],
+    "rust-analyzer.check.invocationLocation": "root",
+    "rust-analyzer.check.invocationStrategy": "once",
+    "rust-analyzer.check.overrideCommand": [
+        "env",
+        "MIRI_AUTO_OPS=no",
+        "./miri",
+        "clippy",
+        "--message-format=json",
+    ],
+    "rust-analyzer.cargo.buildScripts.invocationLocation": "root",
```

Comment:
```suggestion
    // Contrary to what the name suggests, this also affects proc macros.
    "rust-analyzer.cargo.buildScripts.invocationLocation": "root",
```

Reply:
Comments aren't supported by JSON officially, so even when it was in a code block, these lines had a red background suggesting that it's invalid.

Reply:
It works in vscode though, so IMO that's a bug in github's JSON rendered and we should ignore it.

Reply:
That's not a bug, but a feature (: Some parsers go against the specs and allow comments in JSON.

Reply:
What's relevant here is the vscode parser, and it (reasonably) supports comments. So we should use them -- it is quite useful to be able to explain configuration.

I'm not going to merge this with the comments removed.

Reply:
I already added them.